app/Http/Controllers/DinnerformController.php

```diff
@@ -0,0 +1,160 @@
+     */
+    public function create()
+    {
+        $dinnerformList = Dinnerform::all()->sortByDesc('end');
+
+        return view('dinnerform.admin', ['dinnerformCurrent' => null, 'dinnerformList' => $dinnerformList]);
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request $request
+     * @return \Illuminate\Http\RedirectResponse
+     */
+    public function store(Request $request)
+    {
+        $dinnerform = new Dinnerform();
+        $dinnerform->restaurant = $request->restaurant;
+        $dinnerform->description = $request->description;
+        $dinnerform->url = $request->url;
+        $dinnerform->start = strtotime($request->start);
+        $dinnerform->end = strtotime($request->end);
+
+        if ($dinnerform->end < $dinnerform->start) {
+            Session::flash("flash_message", "You cannot let the dinner form close before it opens.");
+            return Redirect::back();
+        }
+
+        $dinnerform->save();
+
+        Session::flash("flash_message", "Your dinner form at '" . $dinnerform->restaurant . "' has been added.");
+        return Redirect::route('dinnerform::add');
+
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  int $id
+     * @return \Illuminate\Http\Response
+     */
+    public function show($id)
+    {
+        $dinnerform = Dinnerform::findOrFail($id);
+
+        if ($dinnerform->isCurrent()) {
+            return Redirect::away($dinnerform->url);
+        } else {
+            Session::flash("flash_message", "Sorry, you can't order anymore, food is already on its way");
+            return Redirect::route('homepage');
+        }
+    }
+
+    /**
+     * Show the form for editing the specified resource.
+     *
+     * @param  int $id
+     * @return \Illuminate\Contracts\View\Factory|\Illuminate\View\View
+     */
+    public function edit($id)
+    {
+        $dinnerformCurrent = Dinnerform::findOrFail($id);
+        $dinnerformList = Dinnerform::all()->sortByDesc('end');
+
+        return view('dinnerform.admin', ['dinnerformCurrent' => $dinnerformCurrent, 'dinnerformList' => $dinnerformList]);
+    }
+
+    /**
+     * Update the specified resource in storage.
+     *
+     * @param  \Illuminate\Http\Request $request
+     * @param  int $id
+     * @return \Illuminate\Http\RedirectResponse
+     */
+    public function update(Request $request, $id)
+    {
+
+        $dinnerform = Dinnerform::findOrFail($id);
+
+        $changed_important_details = $dinnerform->start->timestamp != strtotime($request->start) || $dinnerform->end->timestamp != strtotime($request->end) || $dinnerform->restaurant != $request->restaurant;
+
+        $dinnerform->restaurant = $request->restaurant;
+        $dinnerform->start = strtotime($request->start);
+        $dinnerform->end = strtotime($request->end);
+        $dinnerform->description = $request->description;
```

Comment:
You can use the Dinnerform::Update() function to pass an array of updated fields